androidartifact/apk_info.go

```diff
@@ -76,6 +81,36 @@ func ParsePackageInfos(aaptOut string, isAAB bool) (string, string, string) {
 		parsePackageField(aaptOut, "versionName")
 }
 
+// GetAppNameFromManifest parses app name from `aapt dump badging` command output.
+func GetAppNameFromManifest(aaptOut string, isAAB bool) string {
+	if !isAAB {
+		return parseAppName(aaptOut)
+	}
+
+	pattern := `label=['"](.*?)['"]`
+	re := regexp.MustCompile(pattern)
+	if matches := re.FindStringSubmatch(aaptOut); len(matches) == 2 {
+		return matches[1]
+	}
+
+	return ""
+}
+
+// GetAppNameFromResources parses app name from `aapt dump resources badging` command output.
+func GetAppNameFromResources(aaptOut string) string {
+	pattern := `['"](.*?)['"]`
+	re := regexp.MustCompile(pattern)
+	matches := re.FindAllStringSubmatch(aaptOut, 2)
+	if len(matches) == 2 {
+		appNameMatch := matches[len(matches)-1]
```

Comment:
At this point you already make sure that `len(matches) == 2` so you can make this more readable by using simply
```
appNameMatch := matches[1]
```